data/summary/types/cgm.go

```diff
@@ -283,19 +321,33 @@ func (s *CGMStats) CalculateSummary() {
 	// currently only supports day precision
 	nextStopPoint := 0
 	nextOffsetStopPoint := 0
-	totalStats := &CGMBucketData{}
-	totalOffsetStats := &CGMBucketData{}
+	totalStats := &CGMTotalStats{}
+	totalOffsetStats := &CGMTotalStats{}
+	dayCounted := false
+	offsetDayCounted := false
 
 	for i := 0; i < len(s.Buckets); i++ {
 		currentIndex := len(s.Buckets) - 1 - i
 
+		// new day, reset day counting flag
+		if i%24 == 0 {
```

Comment:
I'm not as up-to-date with the Buckets as I probably should be, but this code would seem to assume there are no "holes" in the data. That is, that each element of a Bucket covers an hour, and that there not gaps or blanks... There's an element per hour for the total span of the buckets. 

Is that true to the design? I poked a little, but didn't find where blanks or empty values were inserted.

Reply:
no holes, but holes are coming soon with the buckets rework im working on, so much has to change to handle both the buckets moving to their own collection, having variable duration (compressing older data points), and having holes matching where there is no data, it's probably not worth putting too much thought into this section, as the entire bucket handling is being rewritten.

Reply:
Thank you for the clarification.